cowprotocol/cowswap
# feat(hook-store): Add balance diff on hook context

# Summary

This PR adds the balance diff information from the bff simulated bundle endpoint with the swap widget. It also sends the current balance diff to the hook app.

# To Test

Swap widget can get token balance changes on pre-hooks.
1. Run one pre hook that will change your tokens (I recommend the Airdrop hook).
2. Check that your balance of the withdraw hook changed (probably this hook will not be on token list you have to import it) on the swap widget components

Swap widget isn't triggering simulations when there is no hook or in other swap types (limit, swap, etc).
1. Go to one tab, play around and check if any `simulateBundle` post request is triggered


Hook Apps can access the current balance diff.
1. Run the app locally
2. Open the build hook on a scenario where there should be a balance diff informed for the hook.
3. Check the console if the balance differences are correct.
